File: src/main/java/org/jcodec/codecs/h264/io/model/RefPicMarkingIDR.java
package org.jcodec.codecs.h264.io.model;

import org.jcodec.common.tools.ToJSON;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * Reference picture marking used for IDR frames
 * 
 * @author Jay Codec
 * 
 */
public class RefPicMarkingIDR {
    boolean discardDecodedPics;
    boolean useForlongTerm;

    public RefPicMarkingIDR(boolean discardDecodedPics, boolean useForlongTerm) {
        this.discardDecodedPics = discardDecodedPics;
        this.useForlongTerm = useForlongTerm;
    }

    public boolean isDiscardDecodedPics() {
        return discardDecodedPics;
    }

    public boolean isUseForlongTerm() {
        return useForlongTerm;
    }
    
    @Override
    public String toString() {
        return ToJSON.toJSON(this);
    }
}

File: src/main/java/org/jcodec/codecs/h264/io/model/RefPicMarkingIDR.kt
package org.jcodec.codecs.h264.io.model

import org.jcodec.platform.Platform

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * Reference picture marking used for IDR frames
 *
 * @author The JCodec project
 */
class RefPicMarkingIDR(var isDiscardDecodedPics: Boolean, var isUseForlongTerm: Boolean) {

    override fun toString(): String {
        return Platform.toJSON(this)
    }

}